### PR TITLE
Feature - 5366 - Changes pagination `pageSizes` options

### LINKED ISSUE #5366: Bump maximum page size to 500
DCM would like to download larger datasets from the application.  To facilitate this we will bump the maximum page size of the paginator to 500 rows.

Context: #5364

Acceptance Criteria
- [x] Bump paginator maximum page size to 500 rows.

### PR DESCRIPTION
**Pull request by mnigh**
🤖 Resolves #5366.

## 👋 Introduction

This PR changes the pagination options for items per page from `10, 20, 30, 40, 50` to `10, 20, 50, 100, 500`.

## 🕵️ Details

Technically speaking, a user can already control the items per page via a url query parameter (ex. `/admin/users?pageSize=1000`) though there is a limit on two of the queries:

### `usersPaginated`
https://github.com/GCTC-NTGC/gc-digital-talent/blob/aa89339bf525d760050427d7da7057ffe03798c6/api/graphql/schema.graphql#L812

### `poolCandidatesPaginated`
https://github.com/GCTC-NTGC/gc-digital-talent/blob/aa89339bf525d760050427d7da7057ffe03798c6/api/graphql/schema.graphql#L866

## 🧪 Testing

1. Navigate to `/admin/pools`
2. Interact with the pagination select element in the table footer
3. Ensure it functions as expected
4. Navigate to `/admin/users`
5. Interact with the pagination select element in the table footer
6. Ensure it functions as expected

## 📸 Screenshot

Screen Shot 2023-01-26 at 16 24 02

### CODE REVIEW

**Review by vd1992 — CHANGES_REQUESTED** [review]
uh-oh

this query (total 189 users selected)

image

results in

image

**Comment by brindasasi** [comment]
https://laravel.com/docs/9.x/queries#chunking-results this might help ?

**Comment by petertgiles** [comment]
Solution: try to solve lazy-load then re-evaluate issue.

**Comment by esizer** [comment]
I'm still running out of memory when I attempt to select all users when having a page size of 500 😢 

```sh
[2023-01-30 17:04:32] local.INFO: GraphQL Error: Attempted to lazy load [awardExperiences] on model [App\Models\User] but lazy loading is disabled.  
[2023-01-30 17:04:32] local.ERROR: Allowed memory size of 134217728 bytes exhausted (tried to allocate 4198400 bytes) {"userId":"222ca041-3842-4b29-91b1-ed47bd01b2b6","exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Allowed memory size of 134217728 bytes exhausted (tried to allocate 4198400 bytes) at /home/site/wwwroot/api/vendor/laravel/framework/src/Illuminate/Http/Response.php:106)
```

File: api/graphql/schema.graphql [hunk 1]
```diff
@@ -269,7 +269,7 @@ type Applicant {
   positionDuration: [PositionDuration] @rename(attribute: "position_duration")
 
   # Experiences
-  experiences: [Experience]
+  experiences: [Experience] @with(relation: "awardExperiences") @with(relation: "communityExperiences") @with(relation: "educationExperiences") @with(relation: "personalExperiences") @with(relation: "workExperiences")
```
[the remark this answers is not in the document]
Comment (mnigh) [reply api/graphql/schema.graphql:272]: Oops, I thought that wasn't needed but it definitely is.

[Add hasMany directive for Applicant type *-Experiences](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5517/commits/451740c14dfcbe3fe8ee3d1ee273fab9fef605ac)

[the remark this answers is not in the document]
Comment (petertgiles) [reply api/graphql/schema.graphql:272]: Probably worth adding the `with`s on the User type as well?

[the remark this answers is not in the document]
Comment (mnigh) [reply api/graphql/schema.graphql:272]: Ok, that makes sense.

[Add with directives for User type experiences](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5517/commits/e55fd86bf75ac1aeef15b590bd30418c02b8924c)

What about here `Skill` type, should they be added there as well?
https://github.com/GCTC-NTGC/gc-digital-talent/blob/b2820b3644aa1335a2a909c43825e905925dcf72/api/graphql/schema.graphql#L587

**Comment by vd1992** [comment]
no more out of memory for 190 users
noticing the selectedUsers operations gets quite slower if you try and grab larger and larger user sets, something to watch out for later

are there any other concerns present with this now?

**Review by vd1992 — APPROVED** [review]
my issue is resolved, nice!
don't know if anyone else has some concerns that merit holding things up

**Comment by mnigh** [comment]
> are there any other concerns present with this now?

just waiting on feedback from @petertgiles on [this question](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5517#discussion_r1090970048) and then once  would be great for @esizer and @petertgiles to re-review to make sure the changes to the schema haven't 💥 anything unintentional.

File: api/graphql/schema.graphql (hunk 1 above)
[the remark this answers is not in the document]
Comment (petertgiles) [reply api/graphql/schema.graphql:272]: I think it probably will but it's out of scope for this issue.  If you're up for it, tag a quick issue on the board for it.

[the remark this answers is not in the document]
Comment (mnigh) [reply api/graphql/schema.graphql:272]: Stub issue created: https://github.com/GCTC-NTGC/gc-digital-talent/issues/5529

**Comment by petertgiles** [comment]
I've seed 500 users into my database and when I try to select them all I'm getting a 500 with no error message.  I'm going to poke at it a bit and see if I can figure out where's it's coming from and if it's something we can reasonably address this sprint.

**Comment by vd1992** [comment]
> I've seed 500 users into my database and when I try to select them all I'm getting a 500 with no error message. I'm going to poke at it a bit and see if I can figure out where's it's coming from and if it's something we can reasonably address this sprint.

when I tried to push user select count to 350, it was failing on me due to the operation taking so long laravel cancelled it (30 second timeout)
which is why I raised the point above, but figured things are slower on my device so it may not be a prod problem anytime soon

**Comment by petertgiles** [comment]
> when I tried to push user select count to 350, it was failing on me due to the operation taking so long laravel cancelled it (30 second timeout)

:laughing: Yeah, mine managed to run out of memory before 30 seconds. 
> FastCGI sent in stderr: "PHP message: PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) in /home/site/wwwroot/api/vendor/laravel/framework/src/Illuminate/Database/Connection.php on line 421

I'm inclined to say we've pushed this one as far as we reasonably can in this issue.

**Review by petertgiles — APPROVED** [review]
:+1:

**Review by esizer — APPROVED** [review]
Going to side with @petertgiles here. I think the selection can wait until a different issue.

**Comment by petertgiles** [comment]
I think the next stop on this path would be to switch some sort of "we'll notify you when your file is ready to be downloaded" system.  And I'm not sure we want to go there...

**Pull request merged**